packages/protocol/contracts/common/UniswapFeeHandlerSeller.sol

```diff
@@ -0,0 +1,194 @@
+pragma solidity ^0.5.13;
+
+import "../common/interfaces/IFeeHandlerSeller.sol";
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+import "../stability/interfaces/IExchange.sol";
+import "../stability/interfaces/ISortedOracles.sol";
+import "./UsingRegistry.sol";
+import "../stability/StableToken.sol";
+import "../common/FixidityLib.sol";
+import "openzeppelin-solidity/contracts/token/ERC20/IERC20.sol";
+import "../common/Initializable.sol";
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+import "openzeppelin-solidity/contracts/math/Math.sol";
+import "./FeeHandlerSeller.sol";
+
+import "../uniswap/interfaces/IUniswapV2RouterMin.sol";
+import "../uniswap/interfaces/IUniswapV2FactoryMin.sol";
+
+contract UniswapFeeHandlerSeller is IFeeHandlerSeller, FeeHandlerSeller {
+  using SafeMath for uint256;
+  using FixidityLib for FixidityLib.Fraction;
+
+  mapping(address => address[]) public routerAddresses;
+  uint256 constant MAX_TIMESTAMP_BLOCK_EXCHANGE = 20;
+
+  event ReceivedQuote(address router, uint256 quote);
+  event RouterUsed(address router);
+  event RouterAddressSet(address token, address router);
+  event RouterAddressRemoved(address token, address router);
+
+  /**
+   * @notice Sets initialized == true on implementation contracts.
+   * @param test Set to true to skip implementation initialisation.
+   */
+  constructor(bool test) public Initializable(test) {}
+
+  // without this line the contract can't receive native Celo transfers
+  function() external payable {}
+
+  /**
+   * @notice Returns the storage, major, minor, and patch version of the contract.
+   * @return Storage version of the contract.
+   * @return Major version of the contract.
+   * @return Minor version of the contract.
+   * @return Patch version of the contract.
+   */
+  function getVersionNumber() external pure returns (uint256, uint256, uint256, uint256) {
+    return (1, 1, 0, 0);
+  }
+
+  /**
+    * @notice Allows owner to set the router for a token.
+    * @param token Address of the token to set.
+    * @param router The new router.
+    */
+  function setRouter(address token, address router) external onlyOwner {
+    _setRouter(token, router);
+  }
+
+  function _setRouter(address token, address router) private {
+    require(router != address(0), "Router can't be address zero");
```

Comment:
This function allows to set same router multiple times. We could use AddressSet instead of array which would solve this issue.